tests/test_config.py

```diff
@@ -0,0 +1,178 @@
+"""
+This module contains unit-tests for config file and CLI argument parsing
+"""
+from io import StringIO
+from textwrap import dedent
+from itertools import permutations
+
+import pytest
+from vulture.config import (
+    DEFAULTS,
+    _parse_args,
+    _parse_toml,
+    from_dict,
+    make_config,
+)
+
+#: This dictionary maps types to "incompatible" types in the config. It is used
+#: later for parametrizing a test.
+WRONG_TYPES = dict(permutations([int, str, list, bool], 2))
```

Comment:
This only yields `{<class 'int'>: <class 'bool'>, <class 'str'>: <class 'bool'>, <class 'list'>: <class 'bool'>, <class 'bool'>: <class 'list'>}` and not all pairwise combinations of types.

Reply:
I'm currently working on this and personally don't like the complexity this introduces into the unit-tests. Especially the "parametrize" logic. I can simplify it and move the complexity from the `parametrize` setup *into* the unit-test but would break the "one check per test" rule in a way.

I feel 50/50 about going into so much details testing this. It's not like this is critical user-input that could generate code-injections or the like, and if a value has the wrong type it will crash out anyway on run.

I will postpone this for now and look at the other issues, let me know your decision on these tests.

Reply:
I agree that I steered you into the wrong direction there, sorry. Let's keep things simple as you originally proposed and just test some manual combinations of bug-prone type pairs.